src/EditorFeatures/CSharp/Highlighting/KeywordHighlighters/AsyncAwaitHighlighter.cs

```diff
@@ -26,7 +28,10 @@ protected override bool IsHighlightableNode(SyntaxNode node)
         protected override IEnumerable<TextSpan> GetHighlightsForNode(SyntaxNode node, CancellationToken cancellationToken)
         {
             var spans = new List<TextSpan>();
-            HighlightRelatedKeywords(node, spans);
+            foreach (var child in node.DescendantNodesAndSelf(n => !n.IsReturnableConstruct()))
```

Comment:
💭 Are we sure `descendIntoChildren` doesn't slightly change the behavior here? Previously we would evaluate this before starting to process a node, but now we process it before processing the children of a node.

Reply:
Will validate.

Reply:
You're correct.  The semantics slightly changed in an observable fashion.  I've updated the PR accordingly.